source/particles/particle_handler.cc

```diff
@@ -380,8 +377,8 @@ namespace Particles
           break;
 
         // Skip cells where there is nothing to remove
-        if (particles_to_remove[n_particles_removed]->active_cell_index !=
-            cell_index)
+        if (particles_to_remove[n_particles_removed]
+              ->cell->active_cell_index() != cell_index)
```

Comment:
What if we now change the loop over cells in line 372 into a loop over iterators? e.g. `for (auto container = particles.begin();...)`. Then we could replace this line with `if (particles_to_remove[n_particles_removed]->particles_on_cell != container)` instead of calling the somewhat expensive `active_cell_index()` here and below in line 390.